pil-analyzer/src/type_inference.rs

```diff
@@ -729,8 +737,30 @@ impl TypeChecker {
                 self.local_var_types.truncate(original_var_count);
                 result?
             }
-            Expression::StructExpression(_sr, _struct_expr) => {
-                unimplemented!("Struct expressions are not yet supported")
+            Expression::StructExpression(sr, StructExpression { name, fields }) => {
+                let Reference::Poly(PolynomialReference { name, .. }) = name else {
+                    unreachable!()
+                };
+                let struct_decl = self.struct_declarations.get(name).unwrap();
+
+                let field_types: HashMap<_, _> = fields
```

Comment:
Is it helpful to create this hash map?

Reply:
Helps to avoid borrowing problems with the `&mut self` of `expect_type`

Reply:
Hm, I don't get it. `type_of_field` returns no references. Where do you borrow `self`?

Reply:
Ah, `struct_decl`?

Reply:
Ok!